util/dvsim/doc/testplanner.md

```diff
@@ -0,0 +1,285 @@
+shared testplan for which substitution values are to be provided in the
+DUT testplan.  When the tool parses the DUT testplan along with the
+imported testplans, it substitutes the wildcards with the substition
+values found in the DUT testplan.  If substitution is not available, then
+the wildcard is replaced with an empty string.  In the example above,
+the list of written test resolves to `["uart_csr_hw_reset"]` after
+substituting `{name}` with `uart` and `{intf}` with an empty string.
+As many wildcards as needed can be added to the tests in the shared
+testplans to support as wide usecases as possible across different
+testbenches. Moreover, the substitution value can be a list of strings,
+in which case, the list of written tests will resolve to all values
+being substituted. See example below for illustration:
+
+```hjson
+-------
+  // Chip testplan:
+  name: chip
+  intf: ["", "_jtag"]
+  foo: ["x", "y", "z"]
+
+-------
+  // Imported testplan:
+  {
+    name: csr
+    ...
+    tests: ["{name}{intf}_csr_hw_reset_{foo}"]
+  }
+```
+
+This will resolve to the following 6 tests:
+
+```
+["chip_csr_hw_reset_x", "chip_csr_hw_reset_y", "chip_csr_hw_reset_z",
+ "chip_jtag_csr_hw_reset_x", "chip_jtag_csr_hw_reset_y", "chip_jtag_csr_hw_reset_z"]
+```
+
+### Example sources
+
+The following examples provided within `util/dvsim/examples/testplanner` can be
+used as a starting point.
+* **`foo_testplan.hjson`**: DUT testplan
+* **`common_testplan.hjson`**: shared testplan imported within the DUT testplan
+* **`foo_dv_doc.md`**: DUT testplan imported within the DV document doc in
+  Markdown
+
+In addition, see the [UART DV document]({{< relref "hw/ip/uart/doc/dv" >}}) for
+a 'production' example of inline expansion of an imported testplan as a table
+within the DV document.
+The [UART testplan](https://github.com/lowRISC/opentitan/blob/master/hw/ip/uart/data/uart_testplan.hjson)
+imports the shared testplans located at `hw/dv/tools/dvsim/testplans` area.
+
+### Limitations
+
+The following limitations currently hold:
+* Only the DUT testplan can import shared testplans; the imported
+  testplans cannot further import more testplans
+* All planned test names parsed from the DUT testplan and all of
+  its imported tetsplans need to be unique
+
+## Usage examples
+
+### Standalone tool invocations
+
+Generate the testplan table in HTML to stdout:
+```console
+$ cd $REPO_TOP
+$ ./util/dvsim/testplanner.py \
+    util/dvsim/examples/testplanner/foo_testplan.hjson
+```
+
+Generate the testplan table in HTML to a file:
+```console
+$ ./util/dvsim/testplanner.py \
+    util/dvsim/examples/testplanner/foo_testplan.hjson \
+    -o /tmp/foo_testplan_table.html
+```
+
+Generate simulation result tables in HTML to stdout:
+```console
+$ ./util/dvsim/testplanner.py \
+    util/dvsim/examples/testplanner/foo_testplan.hjson \
+    -s util/dvsim/examples/testplanner/foo_sim_results.hjson
+```
+
+Generate simulation result tables in HTML to a file:
```

Comment:
Could you comment on where the sim_results.hjson file will be found in actual runs?